Development/nmos-cpp-node/node_implementation.cpp

```diff
@@ -1269,6 +1392,231 @@ nmos::channelmapping_activation_handler make_node_implementation_channelmapping_
+
+    // Each Constraint Set in Sender Caps should contain all parameter constraints from Sender's /constraints/supported
+    auto video_sender_capabilities = value_of({
+        value_of({
+            { nmos::caps::format::media_type, nmos::make_caps_string_constraint({ video_type }) },
+            { nmos::caps::format::grain_rate, nmos::make_caps_rational_constraint({ frame_rate }) },
+            { nmos::caps::format::frame_width, nmos::make_caps_integer_constraint({ frame_width }) },
+            { nmos::caps::format::frame_height, nmos::make_caps_integer_constraint({ frame_height }) },
+            { nmos::caps::format::color_sampling, nmos::make_caps_string_constraint({ color_sampling }) },
+            { nmos::caps::format::interlace_mode, nmos::make_caps_string_constraint({ interlace_mode.name }) },
+            { nmos::caps::format::colorspace, nmos::make_caps_string_constraint({ colorspace }) },
+            { nmos::caps::format::transfer_characteristic, nmos::make_caps_string_constraint({ transfer_characteristic }) },
+            { nmos::caps::format::component_depth, nmos::make_caps_integer_constraint({ component_depth }) },
+            { nmos::caps::transport::st2110_21_sender_type, nmos::make_caps_string_constraint({ sdp::type_parameters::type_N.name }) }
+        })
+    });
+
+    auto audio_sender_capabilities = value_of({
+        value_of({
+            { nmos::caps::format::media_type, nmos::make_caps_string_constraint({ nmos::media_types::audio_L(24).name }) },
+            { nmos::caps::format::channel_count, nmos::make_caps_integer_constraint({ channel_count }) },
+            { nmos::caps::format::sample_rate, nmos::make_caps_rational_constraint({ nmos::rational{48000, 1} }) },
+            { nmos::caps::format::sample_depth, nmos::make_caps_integer_constraint({ 24 }) }
+        })
+    });
+
+    return [&gate, video_sender_capabilities, audio_sender_capabilities, video_sender_ids, audio_sender_ids](const nmos::resource& streamcompatibility_sender, const web::json::value& active_constraints, web::json::value& intersection)
+    {
+        const auto& constraint_sets = nmos::fields::constraint_sets(active_constraints).as_array();
+
+        if (web::json::empty(constraint_sets))
+        {
+            intersection = web::json::value::array();
+            return true;
+        }
+
+        const auto& sender_id = streamcompatibility_sender.id;
+        const bool video_found = video_sender_ids.end() != boost::range::find(video_sender_ids, sender_id);
+        const bool audio_found = audio_sender_ids.end() != boost::range::find(audio_sender_ids, sender_id);
+
+        const auto& sender_capabilities_ = video_found ? video_sender_capabilities : audio_found ? audio_sender_capabilities : throw std::logic_error("No Sender Capabilities found");
+        const auto& sender_capabilities = sender_capabilities_.as_array();
+
+        std::vector<web::json::value> v(constraint_sets.size() * sender_capabilities.size());
```

Comment:
In general I prefer to use `std::vector::reserve` and then use `std::back_inserter` instead of constructing max size and (after setting the values) removing the unused ones by resizing (or erasing up to end).

Here I'm not sure I'd even bother with reserving the capacity because we probably hope/expect that many of the N * M constraint set intersections are `null`, right?

Reply:
I agree. Removed preemptive reserving.